File: rustbridge-csharp/RustBridge.Core/PluginException.cs
namespace RustBridge;

/// <summary>
/// Exception thrown by RustBridge plugin operations.
/// </summary>
public class PluginException : Exception
{
    /// <summary>
    /// Get the error code.
    /// </summary>
    public int ErrorCode { get; }

    /// <summary>
    /// Create a new PluginException.
    /// </summary>
    /// <param name="message">The error message.</param>
    public PluginException(string message) : base(message)
    {
        ErrorCode = 0;
    }

    /// <summary>
    /// Create a new PluginException with an error code.
    /// </summary>
    /// <param name="errorCode">The error code.</param>
    /// <param name="message">The error message.</param>
    public PluginException(int errorCode, string message) : base(message)
    {
        ErrorCode = errorCode;
    }

    /// <summary>
    /// Create a new PluginException with a cause.
    /// </summary>
    /// <param name="message">The error message.</param>
    /// <param name="innerException">The underlying cause.</param>
    public PluginException(string message, Exception? innerException) : base(message, innerException)
    {
        ErrorCode = 0;
    }

    /// <summary>
    /// Create a new PluginException with an error code and cause.
    /// </summary>
    /// <param name="errorCode">The error code.</param>
    /// <param name="message">The error message.</param>
    /// <param name="innerException">The underlying cause.</param>
    public PluginException(int errorCode, string message, Exception? innerException) : base(message, innerException)
    {
        ErrorCode = errorCode;
    }
}

File: rustbridge-java/rustbridge-core/src/main/java/com/rustbridge/PluginException.java
package com.rustbridge;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * Exception thrown by rustbridge plugin operations.
 */
public class PluginException extends Exception {
    private static final long serialVersionUID = 1L;

    private final int errorCode;

    /**
     * Create a new PluginException.
     *
     * @param message the error message
     */
    public PluginException(@NotNull String message) {
        super(message);
        this.errorCode = 0;
    }

    /**
     * Create a new PluginException with an error code.
     *
     * @param errorCode the error code
     * @param message   the error message
     */
    public PluginException(int errorCode, @NotNull String message) {
        super(message);
        this.errorCode = errorCode;
    }

    /**
     * Create a new PluginException with a cause.
     *
     * @param message the error message
     * @param cause   the underlying cause
     */
    public PluginException(@NotNull String message, @Nullable Throwable cause) {
        super(message, cause);
        this.errorCode = 0;
    }

    /**
     * Get the error code.
     *
     * @return the error code, or 0 if not set
     */
    public int getErrorCode() {
        return errorCode;
    }
}
